pkg/scheduler/framework/interface.go

```diff
@@ -704,11 +705,11 @@ func (ni *NominatingInfo) Mode() NominatingMode {
 type PodNominator interface {
 	// AddNominatedPod adds the given pod to the nominator or
 	// updates it if it already exists.
-	AddNominatedPod(pod *PodInfo, nominatingInfo *NominatingInfo)
+	AddNominatedPod(logger klog.Logger, pod *PodInfo, nominatingInfo *NominatingInfo)
```

Comment:
I think pass in context is more expandable for interface.

Reply:
Pass the context is also ok.

Reply:
> I think pass in context is more expandable for interface.

I agree that passing context in an interface is more extensible, but passing context has one problem, as explained below.

I tried passing the context in the following interface:
```
type Cache interface
type SchedulingQueue interface 
type PodNominator interface
```

I found that eventually I needed to modify the resource events handler, such as


			Handler: cache.ResourceEventHandlerFuncs{
				AddFunc:    sched.addPodToSchedulingQueue,
				UpdateFunc: sched.updatePodInSchedulingQueue,
				DeleteFunc: sched.deletePodFromSchedulingQueue,
				AddFunc: func(obj interface{}) {
					sched.addPodToSchedulingQueue(ctx, obj)
				},
				UpdateFunc: func(oldObj, newObj interface{}) {
					sched.updatePodInSchedulingQueue(ctx, oldObj, newObj)
				},
				DeleteFunc: func(obj interface{}) {
					sched.deletePodFromSchedulingQueue(ctx, obj)
				},
			},



The context is always going to be the same in the same scheduler instance. Better look up the logger once and pass that.

In other words,  each time a new event is received, we need to get the logger from the context, which is more expensive than getting the `sched.logger` directly at one time.

Compare details about passing the logger to passing the context: https://github.com/kubernetes/kubernetes/compare/234db6b0479815ce97666eedf26784c874461c25..4b885627199e83cc0a326f0a99189c1759d5bb3f

@alculquicondor  @kerthcet 

ps: I can also change back to pass logger if needed.

Reply:
> I think pass in context is more expandable for interface.

I agree that passing context in an interface is more extensible, but passing context has one problem, as explained below.

I tried passing the context in the following interface:
```
type Cache interface
type SchedulingQueue interface 
type PodNominator interface
```

I found that eventually I needed to modify the resource event handlers, such as


			Handler: cache.ResourceEventHandlerFuncs{
				AddFunc:    sched.addPodToSchedulingQueue,
				UpdateFunc: sched.updatePodInSchedulingQueue,
				DeleteFunc: sched.deletePodFromSchedulingQueue,
				AddFunc: func(obj interface{}) {
					sched.addPodToSchedulingQueue(ctx, obj)
				},
				UpdateFunc: func(oldObj, newObj interface{}) {
					sched.updatePodInSchedulingQueue(ctx, oldObj, newObj)
				},
				DeleteFunc: func(obj interface{}) {
					sched.deletePodFromSchedulingQueue(ctx, obj)
				},
			},



The context is always going to be the same in the same scheduler instance. Better look up the logger once and pass that.

In other words,  each time a new event is received, we need to get the logger from the context, which is more expensive than passing logger directly at one time.

Compare details about passing the logger to passing the context: https://github.com/kubernetes/kubernetes/compare/234db6b0479815ce97666eedf26784c874461c25..4b885627199e83cc0a326f0a99189c1759d5bb3f

ref to：https://github.com/kubernetes/kubernetes/pull/116849#discussion_r1198655475

@alculquicondor  @kerthcet 

ps: I can also change back to pass logger if needed.

Reply:
Yes, pass logger wherever possible, as it's more performant.

Reply:
As @alculquicondor , let's revert to use logger, although it's a little ugly. And sorry for that.

Reply:
Don't be sorry, just for the better~

I have reverted to using logger.